x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/IndexLifecycleOriginationDateParser.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.time.DateFormatter;
+
+import java.util.regex.Pattern;
+
+import static org.elasticsearch.xpack.core.ilm.LifecycleSettings.LIFECYCLE_ORIGINATION_DATE;
+import static org.elasticsearch.xpack.core.ilm.LifecycleSettings.LIFECYCLE_PARSE_ORIGINATION_DATE;
+
+public class IndexLifecycleOriginationDateParser {
+
+    private static final DateFormatter DATE_FORMATTER = DateFormatter.forPattern("yyyy.MM.dd");
+    private static final String INDEX_NAME_REGEX = "^.*-(.+)(-[\\d]+)?$";
+    private static final Pattern INDEX_NAME_PATTERN = Pattern.compile(INDEX_NAME_REGEX);
+
+    /**
+     * Determines if the origination date needs to be parsed from the index name.
+     */
+    public static boolean shouldParseIndexName(Settings indexSettings) {
```

Comment:
I think this should throw an exception when `origination_date` is set to a value and `parse_origination_date` is set to true. We should be strict and reject the index so the user isn't confused about which one takes priority. What do you think?

Reply:
I initially started going down that road (this method was something along the lines of `validateOriginationDateSettings`) but I chose to avoid it as we will set the `origination_date` setting for the index to the date we parse when we initiate the lifecycle policy. If we are to validate the settings this way, our code is creating an invalid setting combination (which we could potentially address by switching the parse setting to `false` when we set the `origination_date` but I didn't like this as we're changing something the user has explicitly set).

I don't have a strong opinion here, but to me, the semantic of an explicit setting overriding what could be parsed was clearer than changing what the user configured.

Reply:
You make a good point about us internally creating settings that have an invalid combination. Let's leave this as-is then.